### DISCUSSION TITLE
Migration documentation improvement

### DISCUSSION THREAD

**Discussion by kevinrenskers** [sveltejs/kit#6146] [General]
The migration guide says this:

> You will need to do this before installing the new version! We recommend updating to the most recent version before these changes — @sveltejs/kit@1.0.0-next.405 — and ensuring that your app works with that version before running the migration.

I am not sure why you need to run the migration guide on 1.0.0-next.405. The effect of that is that as soon as you try to run the app to see if your routes are working, nothing works at all, you just get a 404.

I've seen other people upgrade their version to 1.0.0-next.406 (or later) _before_ running the migration script without any problem. Myself I upgraded immediately after running the migration, which also worked fine. I kinda expected the migration script to update the SvelteKit version to be honest :)

Could the upgrade script be changed to update the SvelteKit version, and then tell you to install it? Or should the docs be changed to tell you upgrade yourself after running the migration script? (And does it really need to be after the script? Doesn't seem like it.)

**Comment by kiliansinger** [comment]
we could also add https://www.loopwerk.io/articles/2022/sveltekit-session-refactor/

without that information it would be pretty hard if you have sessions.

**Reply by kevinrenskers** [reply]
I don't think it makes sense to link out to external articles which might get out of date for example. Also the migration script I was talking about was purely about the routes refactor, not the session refactor :)
